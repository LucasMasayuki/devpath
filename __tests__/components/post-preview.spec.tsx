import React from 'react'
import renderer from 'react-test-renderer'
import PostPreview from '@/components/post-preview'
import Author from '@/types/author'
import Post from '@/types/post'

describe('PostPreview', () => {
  it('should render PostPreview', () => {
    const author: Author = {
      name: 'test',
      picture: 'test',
    }

    const post: Post = {
      content: 'test',
      slug: '',
      title: 'test',
      coverImage: 'test.com',
      date: '1996-10-02',
      author,
      excerpt: 'test',
      ogImage: {
        url: 'test',
      },
    }

    const component = renderer.create(
      <PostPreview
        title={post.title}
        coverImage={post.coverImage}
        date={post.date}
        excerpt={post.excerpt}
        author={post.author}
        slug={post.slug}
      />,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
