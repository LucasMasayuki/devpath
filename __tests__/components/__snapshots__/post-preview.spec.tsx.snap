// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreview should render PostPreview 1`] = `
<div
  className="css-164tvaq"
>
  <div
    className="css-qy636i"
  >
    <img
      alt="Cover Image for test"
      className="chakra-image css-2p2hr9"
      loading="lazy"
      src="test.com"
    />
  </div>
  <div
    className="css-1vae7t2"
  >
    <div
      className="css-j7qwjs"
    >
      <div
        className="css-0"
      >
        <h3
          className="chakra-heading css-1dv4iqg"
        >
          <a
            className="chakra-link css-0"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            test
          </a>
        </h3>
      </div>
      <div
        className="css-1yh2cij"
      >
        <span
          className="chakra-avatar css-1ruhx4w"
        >
          <div
            aria-label="test"
            className="chakra-avatar__initials css-0"
            role="img"
          >
            t
          </div>
        </span>
        <span
          className="css-xf6agq"
        >
          test
        </span>
      </div>
      <div
        className="css-eg4v4b"
      >
        <time
          dateTime="1996-10-02"
        >
          October	2, 1996
        </time>
         - 
        <svg
          className="chakra-icon css-1kmqiqp"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <g
            fill="currentColor"
          >
            <path
              d="M23.432,10.524C20.787,7.614,16.4,4.538,12,4.6,7.6,4.537,3.213,7.615.568,10.524a2.211,2.211,0,0,0,0,2.948C3.182,16.351,7.507,19.4,11.839,19.4h.308c4.347,0,8.671-3.049,11.288-5.929A2.21,2.21,0,0,0,23.432,10.524ZM7.4,12A4.6,4.6,0,1,1,12,16.6,4.6,4.6,0,0,1,7.4,12Z"
            />
            <circle
              cx="12"
              cy="12"
              r="2"
            />
          </g>
        </svg>
        <small>
          0 views
        </small>
      </div>
      <div
        className="css-1fhgjcy"
      >
        <p
          className="chakra-text css-1541wr4"
        >
          test
        </p>
      </div>
    </div>
  </div>
</div>
`;
